src/categorical_algebra/CSets.jl

```diff
@@ -260,4 +260,32 @@ cocone_objects(diagram) = ob(diagram)
 cocone_objects(span::Multispan) = feet(span)
 cocone_objects(para::ParallelMorphisms) = SVector(codom(para))
 
+# Functorial Data migration
+###########################
+
+""" 
+Migrates an instance of one ACSet 
+given dictionaries representing a functor between theories
+"""
+
+function migrate!(Y::ACSet{CD, AD}, X::ACSet, FOb::Dict, FHom::Dict) where {CD, AD}
```

Comment:
I would like to see this be more aggressively typed, but I definitely think we should merge this now, and then when we later have a better Functor type, we can use that.

Reply:
Agreed, there was some discussion of this above.